From 351a1625581517e9d4ea863968b847cc376abeea Mon Sep 17 00:00:00 2001
From: sai surya teja <sssteja@habana.ai>
Date: Fri, 28 Apr 2023 08:53:41 +0300
Subject: [PATCH] [SW-114485] post-image-spi: add preboot die1 pldm hex support

This patch adds preboot die1 pldm hex generation support which
is used for loading hex files on pldm for die1 arcs

Change-Id: I9fd92f081313faf5bbf37711fb5f2f55b553273f
---

diff --git a/board/habanalabs/common/post-image-spi-arc_global.sh b/board/habanalabs/common/post-image-spi-arc_global.sh
index 2299ab2..6d21743 100755
--- a/board/habanalabs/common/post-image-spi-arc_global.sh
+++ b/board/habanalabs/common/post-image-spi-arc_global.sh
@@ -16,6 +16,7 @@
 ARCBOOT_BINFILE=${ARCBOOT_BINFILE:-arcboot_rom.bin}
 PPBOOT_BINFILE=${PPBOOT_BINFILE:-ppboot.bin}
 PREBOOT_BINFILE=${PREBOOT_BINFILE:-preboot.bin}
+PREBOOT_D1_BINFILE=${PREBOOT_D1_BINFILE:-preboot_d1.bin}
 PCIE_PHY_BINFILE=${PCIE_PHY_BINFILE:-pcie_phy_fw.bin}
 PIF_ARC_ELFFILE=${PIF_ARC_FILENAME:-pcie_pif.elf}
 PIF_ICCM_BINFILE=${PIF_ICCM_BINFILE:-pcie_pif-iccm.bin}
@@ -382,6 +383,8 @@
 			hl_pldm_hex preboot${i} x0${i}${PREBOOT_BINFILE}.split.be 1
 		fi
 	done
+	# for preboot_d1 binary
+	prep_pldm_preboot_d1_hex
 
 	# for ROM binary
 	hl_pad_arcboot_rom
diff --git a/board/habanalabs/gaudi2_arc/post-image-spi.sh b/board/habanalabs/gaudi2_arc/post-image-spi.sh
index 96786d6..04dffbd 100755
--- a/board/habanalabs/gaudi2_arc/post-image-spi.sh
+++ b/board/habanalabs/gaudi2_arc/post-image-spi.sh
@@ -540,6 +540,11 @@
 	return
 }
 
+function prep_pldm_preboot_d1_hex {
+	# dummy function, not supported for gaudi2
+	return
+}
+
 function hl_pad_arcboot_rom {
 	#dummy function
 	return
diff --git a/board/habanalabs/gaudi3_arc/post-image-spi.sh b/board/habanalabs/gaudi3_arc/post-image-spi.sh
index 8256bde..de02809 100755
--- a/board/habanalabs/gaudi3_arc/post-image-spi.sh
+++ b/board/habanalabs/gaudi3_arc/post-image-spi.sh
@@ -588,6 +588,11 @@
 	hl_reverse_words ${DEV_PREFIX}-${SPI_TPM_BINFILE} ${DEV_PREFIX}-${SPI_TPM_BINFILE}.le 4
 }
 
+function prep_pldm_preboot_d1_hex {
+	swap_file ${PREBOOT_D1_BINFILE} ${PREBOOT_D1_BINFILE}.orig.be
+	hl_pldm_hex preboot_d1 ${PREBOOT_D1_BINFILE}.orig.be 1
+}
+
 function prep_image_erom {
 	local OUTF=$1
 
